cmd/otel-allocator/benchmark_test.go

```diff
@@ -174,14 +174,20 @@ func createTestDiscoverer(allocationStrategy string, prehookConfig map[string][]
 	allocatorPrehook := prehook.New("relabel-config", logger)
 	allocatorPrehook.SetConfig(prehookConfig)
 	allocator, err := allocation.New(allocationStrategy, logger, allocation.WithFilter(allocatorPrehook))
-	srv := server.NewServer(logger, allocator, "localhost:0")
 	if err != nil {
 		setupLog.Error(err, "Unable to initialize allocation strategy")
 		os.Exit(1)
 	}
+	srv, err := server.NewServer(logger, allocator, "localhost:0")
+	if err != nil {
+		panic(err)
+	}
 	registry := prometheus.NewRegistry()
 	sdMetrics, _ := discovery.CreateAndRegisterSDMetrics(registry)
 	discoveryManager := discovery.NewManager(ctx, config.NopLogger, registry, sdMetrics)
-	targetDiscoverer := target.NewDiscoverer(logger, discoveryManager, allocatorPrehook, srv, allocator.SetTargets)
+	targetDiscoverer, err := target.NewDiscoverer(logger, discoveryManager, allocatorPrehook, srv, allocator.SetTargets)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
I think we should either turn this into a helper function and `require.NoError` here, or (preferable imo) return the error and assert on it in the test function itself.

Reply:
can do